opensrp-reveal/src/main/java/org/smartregister/reveal/util/GeoJsonUtils.java

```diff
@@ -126,67 +82,139 @@ public static String getGeoJsonFromStructuresAndTasks(List<Location> structures,
+                } else if (familyRegTaskMissingOrFamilyRegComplete &&
+                        !state.bednetDistributed && (!state.bloodScreeningDone || (!state.bloodScreeningExists && !state.caseConfirmed))) {
+                    taskProperties.put(TASK_BUSINESS_STATUS, FAMILY_REGISTERED);
+                } else if (state.bednetDistributed && familyRegTaskMissingOrFamilyRegComplete) {
+                    taskProperties.put(TASK_BUSINESS_STATUS, BEDNET_DISTRIBUTED);
+                } else if (state.bloodScreeningDone) {
+                    taskProperties.put(TASK_BUSINESS_STATUS, BLOOD_SCREENING_COMPLETE);
+                } else if (state.ineligibleForFamReg) {
+                    taskProperties.put(TASK_BUSINESS_STATUS, NOT_ELIGIBLE);
+                } else {
+                    taskProperties.put(TASK_BUSINESS_STATUS, NOT_VISITED);
+                }
+
+            } else if (Utils.isMDA()) {
+
+
+                state.fullyReceived = (mdaStatusMap.get(FULLY_RECEIVED).equals(mdaStatusMap.get(MDA_DISPENSE_TASK_COUNT)));
+                state.nonReceived = (mdaStatusMap.get(NONE_RECEIVED).equals(mdaStatusMap.get(MDA_DISPENSE_TASK_COUNT)));
+                state.nonEligible = (mdaStatusMap.get(NOT_ELIGIBLE).equals(mdaStatusMap.get(MDA_DISPENSE_TASK_COUNT)));
+                state.partiallyReceived = (!state.fullyReceived && (mdaStatusMap.get(FULLY_RECEIVED) > 0));
+
+                if (familyRegTaskMissingOrFamilyRegComplete) {
+                    if (state.mdaAdhered) {
+                        taskProperties.put(TASK_BUSINESS_STATUS, ADHERENCE_VISIT_DONE);
+                    } else if (state.fullyReceived) {
+                        taskProperties.put(TASK_BUSINESS_STATUS, FULLY_RECEIVED);
+                    } else if (state.partiallyReceived) {
+                        taskProperties.put(TASK_BUSINESS_STATUS, PARTIALLY_RECEIVED);
+                    } else if (state.nonReceived) {
+                        taskProperties.put(TASK_BUSINESS_STATUS, NONE_RECEIVED);
+                    } else if (state.nonEligible) {
+                        taskProperties.put(TASK_BUSINESS_STATUS, NOT_ELIGIBLE);
                     } else {
-                        taskProperties.put(TASK_BUSINESS_STATUS, NOT_VISITED);
+                        taskProperties.put(TASK_BUSINESS_STATUS, FAMILY_REGISTERED);
                     }
-
+                } else {
+                    taskProperties.put(TASK_BUSINESS_STATUS, NOT_VISITED);
                 }
 
             }
 
-            structure.getProperties().setCustomProperties(taskProperties);
-
         }
-        return gson.toJson(structures);
+    }
+
+    private static class StateWrapper {
+        boolean familyRegistered = false;
+        boolean bednetDistributed = false;
+        boolean bloodScreeningDone = false;
+        boolean familyRegTaskExists = false;
+        boolean caseConfirmed = false;
+        boolean mdaAdhered = false;
+        boolean fullyReceived;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=4684568)